### PR TITLE
feat: Created composite Simpson's numerical integration method

### PR DESCRIPTION
**Pull request by ggkogkou**
#### Description of Change
Created the composite Simpson's numerical method for integrals approximation (wiki: https://en.wikipedia.org/wiki/Simpson%27s_rule#Composite_Simpson's_rule). 

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 
Numerical method for integral evaluation implementation

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1773"></a>

### CODE REVIEW

**Review by ayaankhan98 — CHANGES_REQUESTED** [review]
Thanks for your contribution. Please fix the issue below.
```
test.cpp: In function ‘void test(int, double, double, double, bool)’:
test.cpp:96:87: error: ‘f’ was not declared in this scope; did you mean ‘numerical_methods::simpson_method::f’?
   96 |     double result_f = numerical_methods::simpson_method::evaluate_by_simpson(N, h, a, f);
      |                                                                                       ^
      |                                                                                       numerical_methods::simpson_method::f
test.cpp:68:8: note: ‘numerical_methods::simpson_method::f’ declared here
   68 | double f(double x){
      |        ^
test.cpp:100:87: error: ‘g’ was not declared in this scope; did you mean ‘numerical_methods::simpson_method::g’?
  100 |     double result_g = numerical_methods::simpson_method::evaluate_by_simpson(N, h, a, g);
      |                                                                                       ^
      |                                                                                       numerical_methods::simpson_method::g
test.cpp:72:8: note: ‘numerical_methods::simpson_method::g’ declared here
   72 | double g(double x){
      |        ^
test.cpp:104:87: error: ‘k’ was not declared in this scope; did you mean ‘numerical_methods::simpson_method::k’?
  104 |     double result_k = numerical_methods::simpson_method::evaluate_by_simpson(N, h, a, k);
      |                                                                                       ^
      |                                                                                       numerical_methods::simpson_method::k
test.cpp:76:8: note: ‘numerical_methods::simpson_method::k’ declared here
   76 | double k(double x){
      |        ^
test.cpp:108:87: error: ‘l’ was not declared in this scope; did you mean ‘numerical_methods::simpson_method::l’?
  108 |     double result_l = numerical_methods::simpson_method::evaluate_by_simpson(N, h, a, l);
      |                                                                                       ^
      |                                                                                       numerical_methods::simpson_method::l
test.cpp:80:8: note: ‘numerical_methods::simpson_method::l’ declared here
   80 | double l(double x){
      |        ^

shell returned 1
```

File: numerical_methods/composite_simpson_rule.cpp [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+    // Find the step
+    h = (b-a)/N;
+
+    test(N, h, a, b, used_argv_parameters); /// run self-test implementations
+
+    return 0;
+}
+
+double numerical_methods::simpson_method::evaluate_by_simpson(int N, double h, double a, std::function<double (double)> func){
```
Comment (Panquesito7) [inline numerical_methods/composite_simpson_rule.cpp:150]: Missing function documentation.

File: numerical_methods/composite_simpson_rule.cpp [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+        std::cout << "You selected N=" << N << ", a=" << a << ", b=" << b << std::endl;
+    }
+    else
+        std::cout << "Default N=" << N << ", a=" << a << ", b=" << b << std::endl;
+
+
+    // Find the step
+    h = (b-a)/N;
+
+    test(N, h, a, b, used_argv_parameters); /// run self-test implementations
```
Comment (Panquesito7) [inline numerical_methods/composite_simpson_rule.cpp:145]:
```suggestion
    test(N, h, a, b, used_argv_parameters);  // run self-test implementations
```

**Review by Panquesito7 — COMMENTED** [review]
Everything else is looking great! 🚀 Just fix the errors mentioned by @ayaankhan98 above, please. Let us know if you need any help! 🙂

**Comment by ggkogkou** [comment]
Done

**Review by Panquesito7 — APPROVED** [review]
LGTM 🚀 Please review, @ayaankhan98. Thanks. 🙂

**Review by ayaankhan98 — APPROVED** [review]
:+1:

**Pull request merged**